### PR TITLE
feat: change instance specific dimensions in aggregated metrics

### PR DESCRIPTION
**Pull request by kruskall**
Add service instance transaction metrics and change instance specific dimensions in aggregated metrics

Related to https://github.com/elastic/apm-server/issues/11266

### CODE REVIEW

File: aggregators/merger.go [hunk 1]
```diff
@@ -136,6 +147,37 @@ func (m *combinedMetricsMerger) merge(from *aggregationpb.CombinedMetrics) {
 	}
 }
 
+// mergeTransactionGroups merges service instance transaction aggregation groups for two combined metrics
```
Comment (carsonip) [inline aggregators/merger.go:150]:
```suggestion
// mergeServiceInstanceTransactionGroups merges service instance transaction aggregation groups for two combined metrics
```

File: aggregators/converter.go [hunk 2]
```diff
@@ -287,6 +316,10 @@ func getEventMetricsBuilder(partition uint16) *eventMetricsBuilder {
 	mb.keyedTransactionMetrics.Metrics = &mb.transactionMetrics
 	mb.keyedTransactionMetricsArray[0] = &mb.keyedTransactionMetrics
 	mb.keyedTransactionMetricsSlice = mb.keyedTransactionMetricsArray[:0]
+	mb.keyedServiceInstanceTransactionMetrics.Key = &mb.serviceInstanceTransactionAggregationKey
```
Comment (carsonip) [inline aggregators/converter.go:319]: Test failures are caused by missing reset on line 305
```
mb.keyedServiceInstanceTransactionMetricsSlice = mb.keyedServiceInstanceTransactionMetricsSlice[:0]
```

I hate github code review UI, can't leave comments on unchanged lines.

**Review by carsonip — CHANGES_REQUESTED** [review]

File: aggregators/converter.go [hunk 3]
```diff
@@ -249,6 +271,13 @@ type eventMetricsBuilder struct {
 	keyedServiceTransactionMetricsArray [1]*aggregationpb.KeyedServiceTransactionMetrics
 	keyedServiceTransactionMetricsSlice []*aggregationpb.KeyedServiceTransactionMetrics
 
+	// There can be at most ? service instance transaction metric per event.
```
Comment (lahsivjar) [inline aggregators/converter.go:274]:
```suggestion
	// There can be at most 1 service instance transaction metric per event.
```

The `partitionedMetricsBuilder` will build `CombinedMetrics` for a single `APMEvent`. Thus, there will only be 1 service instance transaction metric per APMEvent.

**Review by lahsivjar — APPROVED** [review]
Code LGTM!

File: aggregators/merger.go [hunk 4]
```diff
@@ -136,6 +147,37 @@ func (m *combinedMetricsMerger) merge(from *aggregationpb.CombinedMetrics) {
 	}
 }
 
+// mergeServiceInstanceTransactionGroups merges service instance transaction aggregation groups for two combined metrics
+// considering max transaction groups and max transaction groups per service limits.
```
Comment (carsonip) [inline aggregators/merger.go:151]:
```suggestion
// considering max service instance transaction groups and max service instance transaction groups per service limits.
```

**Review by carsonip — APPROVED** [review]

**Pull request merged**